packages/catalog-process/src/routers/catalog.ts

```diff
@@ -88,7 +91,51 @@ const eservicesRouter = (
             req.params.descriptorId,
             req.params.documentId,
             req.body,
-            req.authData
+            req.ctx.authData
+          );
+          return res.status(200).end();
+        } catch (error) {
+          const errorRes: ApiError = makeApiError(error);
+          return res.status(errorRes.status).json(errorRes).end();
+        }
```

Comment:
maybe we could extract this `try-catch` structure or at least the `catch` body into a function, since it looks like it's the same for all routes?

Reply:
as discussed offline this method has a bit of complexity with generics types inferred from Zodios types and API definition. @gabro This is the second time that we discuss this generalization, if you have some opinions or suggestions track them in Jira Card, for the moment we postpone it.